spring-pulsar/src/main/java/org/springframework/pulsar/core/reactive/DefaultReactivePulsarSenderFactory.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.core.reactive;
+
+import java.util.List;
+
+import org.apache.commons.logging.LogFactory;
+import org.apache.pulsar.client.api.MessageRouter;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.reactive.client.adapter.AdaptedReactivePulsarClientFactory;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageSender;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageSenderBuilder;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageSenderSpec;
+import org.apache.pulsar.reactive.client.api.ReactivePulsarClient;
+
+import org.springframework.core.log.LogAccessor;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * Default implementation of {@link ReactivePulsarSenderFactory}.
+ *
+ * @param <T> reactive sender type.
+ * @author Christophe Bornet
+ */
+public class DefaultReactivePulsarSenderFactory<T> implements ReactivePulsarSenderFactory<T> {
+
+	private final LogAccessor logger = new LogAccessor(LogFactory.getLog(this.getClass()));
+
+	private final ReactivePulsarClient reactivePulsarClient;
+
+	private final ReactiveMessageSenderSpec reactiveMessageSenderSpec;
+
+	public DefaultReactivePulsarSenderFactory(PulsarClient pulsarClient,
+			ReactiveMessageSenderSpec reactiveMessageSenderSpec) {
```

Comment:
The `ReactiveMessageSenderSpec` is akin to the map of config props in the non-reactive producer factor. And the SB auto-config will ultimately build that up and pass it in, correct?

Reply:
yes